test/dynamo/test_decorators.py

```diff
@@ -1953,34 +1953,85 @@ def f3():
         with self.assertRaises(Exception):
             f3()
 
-    def test_nested_compile_fullgraph(self):
+    def test_nested_compile_error_on_graph_break(self):
         inp = torch.ones(3)
 
-        @torch.compile(backend="eager", fullgraph=True)
+        @torch.compile(backend="eager", error_on_graph_break=True)
         def inner_f1(x):
             x = x + 1
             torch._dynamo.graph_break()
             return x + 2
 
-        @torch.compile(backend="eager", fullgraph=False)
+        @torch.compile(backend="eager", error_on_graph_break=False)
         def f1(x):
             return inner_f1(x)
 
         with self.assertRaises(Unsupported):
             f1(inp)
 
-        @torch.compile(backend="eager", fullgraph=False)
+        @torch.compile(backend="eager", error_on_graph_break=False)
         def inner_f2(x):
             x = x + 1
             torch._dynamo.graph_break()
             return x + 2
 
-        @torch.compile(backend="eager", fullgraph=True)
+        @torch.compile(backend="eager", error_on_graph_break=True)
         def f2(x):
             return inner_f2(x)
 
         self.assertEqual(f2(inp), inp + 3)
 
+    def test_error_on_graph_break_fullgraph(self):
+        inp = torch.ones(3)
```

Comment:
Might be worth commenting that this aims to test the top row of our programming model (interaction with fullgraph=True and error_on_grpah_break values)

I think making that explicit here would be helpful